listenbrainz/webserver/views/profile.py

```diff
@@ -51,6 +54,35 @@ def reset_token():
     )
 
 
+@profile_bp.route("/select_timezone/", methods=["GET", "POST"])
+@login_required
+def select_timezone():
+    pg_timezones = db_usersetting.get_pg_timezone()
+    form = TimezoneForm()
+    form.timezone.choices = [(zone_name, zone_name) for (zone_name, offset) in pg_timezones]
+    user_settings = db_usersetting.get(current_user.id)
+    user_timezone = user_settings['timezone_name']
+    if form.validate_on_submit():
+        try:
+            update_timezone = str(form.timezone.data)
+            db_usersetting.set_timezone(current_user.id, update_timezone)
+            flash.info("timezone reset")
+        except DatabaseException:
+            flash.error("Something went wrong! Unable to update timezone right now.")
+        return redirect(url_for("profile.info"))
+
+    if form.csrf_token.errors:
+        flash.error('Cannot update timezone due to error during authentication, please try again later.')
```

Comment:
For this error, lets give fewer details, since that might mean someone is trying to hack us. If that is the case, lets not give then any help. Lets have this say: "Unable to update timezone."